crates/evm/src/tests/ef_tests/cases/blockchain_test.rs

```diff
@@ -0,0 +1,291 @@
+        Ok(BlockchainTestCase {
+            tests: {
+                let s = fs::read_to_string(path).map_err(|error| Error::Io {
+                    path: path.into(),
+                    error,
+                })?;
+                serde_json::from_str(&s).map_err(|error| Error::CouldNotDeserialize {
+                    path: path.into(),
+                    error,
+                })?
+            },
+            skip: should_skip(path),
+        })
+    }
+
+    /// Runs the test cases for the Ethereum Forks test suite.
+    ///
+    /// # Errors
+    /// Returns an error if the test is flagged for skipping or encounters issues during execution.
+    fn run(&self) -> Result<(), Error> {
+        // If the test is marked for skipping, return a Skipped error immediately.
+        if self.skip {
+            return Err(Error::Skipped);
+        }
+
+        // Iterate through test cases, filtering by the network type to exclude specific forks.
+        self.tests
+            .values()
+            .filter(|case| matches!(case.network, ForkSpec::Shanghai))
+            .par_bridge()
+            .try_for_each(|case| {
+                let mut evm_config = EvmConfig::default();
+
+                // Set this base fee based on what's set in genesis.
+                let header = reth_primitives::Header {
+                    parent_hash: case.genesis_block_header.parent_hash,
+                    ommers_hash: EMPTY_OMMER_ROOT_HASH,
+                    beneficiary: evm_config.coinbase,
+                    // This will be set in finalize_hook or in the next begin_slot_hook
+                    state_root: case.genesis_block_header.state_root,
+                    transactions_root: case.genesis_block_header.transactions_trie,
+                    receipts_root: case.genesis_block_header.receipt_trie,
+                    withdrawals_root: case.genesis_block_header.withdrawals_root,
+                    logs_bloom: case.genesis_block_header.bloom,
+                    difficulty: case.genesis_block_header.difficulty,
+                    number: case.genesis_block_header.number.to(),
+                    gas_limit: case.genesis_block_header.gas_limit.to(),
+                    gas_used: case.genesis_block_header.gas_used.to(),
+                    timestamp: case.genesis_block_header.timestamp.to(),
+                    mix_hash: case.genesis_block_header.mix_hash,
+                    nonce: case.genesis_block_header.nonce.into(),
+                    base_fee_per_gas: case.genesis_block_header.base_fee_per_gas.map(|b| b.to()),
+                    extra_data: case.genesis_block_header.extra_data.clone(),
+                    // EIP-4844 related fields
+                    // https://github.com/Sovereign-Labs/sovereign-sdk/issues/912
+                    blob_gas_used: case.genesis_block_header.blob_gas_used.map(|b| b.to()),
+                    excess_blob_gas: case.genesis_block_header.excess_blob_gas.map(|b| b.to()),
```

Comment:
What do you think of blobs here? @eyusufatik

Reply:
does not really matter if it's there or not